inference_perf/metrics/base.py

```diff
@@ -12,21 +12,24 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 from abc import ABC, abstractmethod
+from typing import Any
 from pydantic import BaseModel
 
 
 class MetricsSummary(BaseModel):
     total_requests: int
-    avg_prompt_tokens: float
-    avg_output_tokens: float
-    avg_time_per_request: float
-
+    avg_prompt_tokens: int
+    avg_output_tokens: int
+    avg_request_latency: float
+    avg_time_to_first_token: float
+    avg_time_per_output_token: float
+    avg_queue_length: int
 
 class MetricsClient(ABC):
     @abstractmethod
     def __init__(self) -> None:
         pass
 
     @abstractmethod
-    def collect_metrics_summary(self) -> MetricsSummary | None:
+    def collect_metrics_summary(self, duration = None, model_server_client = None) -> MetricsSummary | None:
```

Comment:
Lets consolidate these args to a broader `runtime_parameters` that has test duration and model attributes as follows,

```suggestion
    def collect_metrics_summary(self, runtime_parameters: RuntimeParameters = None) -> MetricsSummary | None:
```